arbnode/batch_poster.go

```diff
@@ -431,6 +431,35 @@ func AccessList(opts *AccessListOpts) types.AccessList {
 	return l
 }
 
+type txData struct {
+	Hash  common.Hash    `json:"hash"`
+	Nonce hexutil.Uint64 `json:"nonce"`
+	From  common.Address `json:"from"`
+}
+
+// getTxsDataByBlock fetches all the transactions inside block of id 'number' using json rpc
+// and returns an array of txData which has fields that are necessary in checking for batch reverts
+func (b *BatchPoster) getTxsDataByBlock(ctx context.Context, number int64) ([]txData, error) {
```

Comment:
It might be simpler to use `eth_getBlockByNumber(number, true)` here instead

Reply:
oh yeah, I missed that boolean option! it makes it super helpful to get transactions that are serialized in RPC representation. Thanks